luxtronik/__init__.py

```diff
@@ -103,26 +116,30 @@ def _with_lock_and_connect(self, func, *args, **kwargs):
             ret_val = func(*args, **kwargs)
             return ret_val
 
-    def read(self):
+    def read(self, data=None):
         """
         Read data from heat pump.
         All available data will be read from the heat pump.
         """
-        return self._with_lock_and_connect(self._read)
+        if data is None:
+            data = LuxtronikData.new(self._safe)
+        return self._with_lock_and_connect(self._read, data)
 
-    def read_parameters(self):
+    def read_parameters(self, parameters=None):
         """Read parameters from heat pump."""
-        return self._with_lock_and_connect(self._read_parameters)
+        if parameters is None:
+            parameters = Parameters(self._safe)
```

Comment:
I am not sure if `LuxtronikInterface` should hold the `_safe` flag. I mean, this is not related to the interface itself. I propose that `read_parameters` just create `Parameters()` (with the default `safe = True`) if no `parameters` are passed.

Reply:
Done. Currently, there is still the parameter at `Luxtronik()`